### ISSUE TITLE
One more problem with dense output of the `Solution` class

### ISSUE THREAD

**Issue by sebhoof** [fruzsinaagocs/oscode#15]
This is similar to https://github.com/fruzsinaagocs/oscode/issues/13 in that the dense output variables and hence also the data written to file are incorrect, e.g. set to 0. This only happens when both these conditions are met: (a) `t_i` > `t_f` and (b) the times in `const X &do_times` are also decreasing. 

To reproduce this in a minimal example, I again modified the `burst.cpp file` as follows

```
ti = 2*n;
tf = -2*n;

std::vector<double> temp;
for (int i = 0; i < 1000; ++i) { temp.push_back(2*n - 4*n*i*0.001); }

de_system sys(&w, &g);
Solution solution(sys, x0, dx0, ti, tf, temp, 3, 1e-4, 0.0, 1, "native_do.dat");
solution.solve();

std::vector<double> do_ts(std::begin(solution.dotimes), std::end(solution.dotimes));
std::vector<std::complex<double> > do_xs(std::begin(solution.dosol), std::end(solution.dosol));

for (int i = 0; i < do_xs.size(); ++i)
{
     std::cout << temp[i] << " " << do_ts[i] << " " << do_xs[i] << std::endl;
}
```
I guess one way to solve this is to ensure that the dense output times are always sorted in increasing order, as the issues does not seem to appear then.

**Comment by sebhoof** [comment]
Commit [6b60391](https://github.com/fruzsinaagocs/oscode/commit/6b60391b74c6bbde1bd32809fcaccc4641e772c5) correctly reverses the ordering of `dotimes` if necessary, but not of `dosol` and `dodsol`.

**Comment by sebhoof** [comment]
Sorry, I now realised that this was further modified in [6cd21f1](https://github.com/fruzsinaagocs/oscode/commit/6cd21f1ed0155235c04b964ee31968d470f7c070), but I think that `std::reverse()` needs to be used instead; see commit [cec8379](https://github.com/fruzsinaagocs/oscode/commit/cec8379b2b389b7c64cd75cbce3c14a8ebe36387).

**Comment by fruzsinaagocs** [comment]
I found that `reverse` vs `std::reverse` didn't make a difference for me, but there were other issues:
- The documentation said that no matter the integration order or the order of the supplied dense output points, the output dense output points would always be in ascending order. This was false -- only `dosol` and `dodsol` were obeying this but not `dotimes`. Fixed now.
- The dense output points were written to the output file before any reversal could take place. Fixed now.

I tested this fix by integrating forwards and backwards in time and asking for dense output points in ascending or descending order; it now always sorts dense output times in ascending order (and the dense output solution & derivative are consistent with this).
